packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -148,3 +167,216 @@ contract OPContractsManager_InternalMethods_Test is Test {
+                saltMixer: "hello",
+                gasLimit: 30_000_000,
+                disputeGameType: GameType.wrap(1),
+                disputeAbsolutePrestate: Claim.wrap(
+                    bytes32(hex"038512e02c4c3f7bdaec27d00edf55b7155e0905301e1a88083e4e0a6764d54c")
+                ),
+                disputeMaxGameDepth: 73,
+                disputeSplitDepth: 30,
+                disputeClockExtension: Duration.wrap(10800),
+                disputeMaxClockDuration: Duration.wrap(302400)
+            })
+        );
+    }
+
+    function test_addGameType_permissioned_succeeds() public {
+        OPContractsManager.AddGameInput memory input = newGameInputFactory(true);
+        OPContractsManager.AddGameOutput memory output = addGameType(input);
+        assertValidGameType(input, output);
+        IPermissionedDisputeGame newPDG = IPermissionedDisputeGame(address(output.faultDisputeGame));
+        IPermissionedDisputeGame oldPDG = chainDeployOutput.permissionedDisputeGame;
+        assertEq(newPDG.proposer(), oldPDG.proposer(), "proposer mismatch");
+        assertEq(newPDG.challenger(), oldPDG.challenger(), "challenger mismatch");
+    }
+
+    function test_addGameType_permissionless_succeeds() public {
+        OPContractsManager.AddGameInput memory input = newGameInputFactory(false);
+        OPContractsManager.AddGameOutput memory output = addGameType(input);
+        assertValidGameType(input, output);
+        IPermissionedDisputeGame notPDG = IPermissionedDisputeGame(address(output.faultDisputeGame));
+        vm.expectRevert(); // nosemgrep: sol-safety-expectrevert-no-args
+        notPDG.proposer();
+    }
+
+    function test_addGameType_outOfOrderInputs_reverts() public {
+        OPContractsManager.AddGameInput memory input1 = newGameInputFactory(false);
+        input1.disputeGameType = GameType.wrap(2);
+        OPContractsManager.AddGameInput memory input2 = newGameInputFactory(false);
+        input2.disputeGameType = GameType.wrap(1);
+        OPContractsManager.AddGameInput[] memory inputs = new OPContractsManager.AddGameInput[](2);
+        inputs[0] = input1;
+        inputs[1] = input2;
+
+        // For the sake of completeness, we run the call again to validate the success behavior.
+        (bool success,) = address(opcm).delegatecall(abi.encodeCall(OPContractsManager.addGameType, (inputs)));
+        assertFalse(success, "addGameType should have failed");
+    }
+
+    function test_addGameType_duplicateGameType_reverts() public {
+        OPContractsManager.AddGameInput memory input = newGameInputFactory(false);
+        OPContractsManager.AddGameInput[] memory inputs = new OPContractsManager.AddGameInput[](2);
+        inputs[0] = input;
+        inputs[1] = input;
+
+        // See test above for why we run the call twice.
+        (bool success,) = address(opcm).delegatecall(abi.encodeCall(OPContractsManager.addGameType, (inputs)));
+        assertFalse(success, "addGameType should have failed");
```

Comment:
syntax might be off, but would be ideal to test the revert reason here too (similar for below tests, let's add the reasons for a stronger test)
```suggestion
        // See test above for why we run the call twice.
        (bool success, bytes memory revertData) = address(opcm).delegatecall(abi.encodeCall(OPContractsManager.addGameType, (inputs)));
        assertFalse(success, "addGameType should have failed");
        assertEq(revertData, InvalidGameConfigs.selector);
```